docs/zkapps/tutorials/06-offchain-storage.mdx

```diff
@@ -282,39 +321,51 @@ Continuing, here is our update function:
+This code gets and asserts the current state of the contract, and then performs the update.
 
-First, we check that the new leaf is greater than the old leaf.
+First, check that the new leaf is greater than the old leaf.
 
-Then, we check the update itself. In this example, we perform a single update to the tree, however with multiple witnesses, you can chain updates together to change the tree more than once in a single call to the storage server.
+Then, check the update itself. In this example, perform a single update to the tree. However, you can chain updates together with multiple witnesses to change the tree more than once in a single call to the storage server.
 
-To assert the update is valid, we use a `assertRootUpdateValid` call from the `OffChainStorage` library. This checks that when the update is applied to the tree represented by the existing on-chain tree root, the data for the new tree is being stored by the storage server.
+To assert the update is valid, use a `assertRootUpdateValid` call from the `OffChainStorage` library. This checks that when the update is applied to the tree represented by the existing on-chain tree root, the data for the new tree is being stored by the storage server.
 
 That completes the smart contract!
 
 ## Implementing `main.ts`
 
-You can find a full copy of this file [here](https://github.com/o1-labs/docs2/blob/main/examples/zkapps/06-offchain-storage/offchain-storage-zkapp/contracts/src/main.ts) for reference.
+Since much of the logic in the `main.ts` file is repeated from earlier tutorials, this tutorial reviews just the relevant parts. 
+
+1. Download the [main.ts](https://github.com/o1-labs/docs2/blob/main/examples/zkapps/06-offchain-storage/offchain-storage-zkapp/contracts/src/main.ts) example file.
+
+1. Move it to the local `/06-off-chain-storage/src` folder.
 
-We will not be implementing this full file, instead just discussing a few parts of it, since much is repeated from earlier tutorials. So download it from the above link, place it in your src folder, and then open it in your editor.
+1. Open it in your editor.
 
-Note that it contains logic for both running the contract locally and for deploying and interacting with it on Berkeley. This is a useful pattern when developing a new contract. If you would like to try it on Berkeley, deploy our contract as usual with `zk deploy`, set `useLocal` to false in `main.ts`, and add the name of your config to the end of your call to `node main.js`.
+The `main.ts` file contains logic for running the contract locally and for deploying and interacting with it on Berkeley. 
 
-To start, we will connect to the off-chain storage server:
+This is a useful pattern when developing a new contract. 
+
+### Connect to the off-chain storage server
+
+To try your contract on Berkeley, deploy the contract as usual with `zk deploy`, set `useLocal` to false in `main.ts`, and add the name of your config to the end of your call to `node main.js`.
```

Comment:
@Trivo25  can you help here? what are the specific steps that we are asking our reader to do here? let's provide guided steps and examples